### PR TITLE
Add org.kde.okteta (Okteta)

### PR DESCRIPTION
**Pull request by Siosm**
Initial import from https://invent.kde.org/packaging/flatpak-kde-applications

### CODE REVIEW

File: appdata.patch [hunk 1]
```diff
@@ -0,0 +1,12 @@
+diff --git program/org.kde.okteta.appdata.xml program/org.kde.okteta.appdata.xml
+index beed216b..0347c3aa 100644
+--- a/program/org.kde.okteta.appdata.xml
++++ b/program/org.kde.okteta.appdata.xml
+@@ -553,7 +553,6 @@
+     </screenshot>
+   </screenshots>
+   <project_group>KDE</project_group>
+-  <icon type="stock">okteta</icon>
```
Comment (bilelmoussaoui) [inline appdata.patch:9]: It would be nice to send this patch upstream if possible

**Comment by bilelmoussaoui** [comment]
bot, build org.kde.okteta

File: appdata.patch (hunk 1 above)
Comment (Siosm) [reply appdata.patch:9]: In progess: https://invent.kde.org/utilities/okteta/-/merge_requests/3 :slightly_smiling_face:

**Comment by Siosm** [comment]
Updated the temporary patch.

**Comment by bilelmoussaoui** [comment]
bot, build org.kde.okteta

**Comment by bilelmoussaoui** [comment]
This probably needs a rename-icon on the manifest

**Comment by Siosm** [comment]
Mention @tsdgeos as he currently is a maintainer of several KDE related packages on Flathub.

**Comment by Siosm** [comment]
Anyway I could run all those checks beforehand?

**Comment by Siosm** [comment]
Fixed

**Comment by bilelmoussaoui** [comment]
bot, build org.kde.okteta

You can trigger a new build as well by the way

**Comment by bilelmoussaoui** [comment]
The project seems to include only a 16, 32, 48px of the icon https://invent.kde.org/utilities/okteta/-/blob/master/program/CMakeLists.txt#L45-48
We do require a 128px if possible; you can just up-scale the icon for now

**Comment by Siosm** [comment]
bot, build org.kde.okteta

**Comment by Siosm** [comment]
This one is stuck as it does not like binary git patches.

**Comment by Siosm** [comment]
bot, build org.kde.okteta

**Comment by tsdgeos** [comment]
Add the device dri thing?

**Comment by Siosm** [comment]
bot, build org.kde.okteta

**Comment by tsdgeos** [comment]
Needs
--socket=cups
so that the printers show in the print dialog (if you're using Plasma)

**Comment by Siosm** [comment]
bot, build org.kde.okteta

**Comment by tsdgeos** [comment]
Looks good to me :)

**Comment by travier** [comment]
I will do a general conversion to yaml & file cleanup soon.

**Comment by Siosm** [comment]
I will do a general conversion to yaml & file cleanup soon for all the PRs.

**Comment by tsdgeos** [comment]
No, do not convert to yaml

**Comment by Siosm** [comment]
Any particular reason why I should not convert everything to yaml?

**Comment by tsdgeos** [comment]
> Any particular reason why I should not convert everything to yaml?

Becasue the existing > 30 org.kde. things are in json, having everything in the same syntax helps

**Comment by Siosm** [comment]
I will push PRs to convert all of those (and the nightly ones too) to YAML if you are OK.

**Comment by tsdgeos** [comment]
No, please do not do that. I have tools that go over the json file and update it automatically (so that i don't have to change 20.04.1 to 20.04.2 + sha in 30+ repos) if you do that, you'll break me

**Comment by Siosm** [comment]
I was planning on building such a tool too so maybe we can share the load here. Would you be OK to share those tools so that I can update them to use YAML? I was also pointed to https://invent.kde.org/jbbgameich/flatpak-kde-applications/-/blob/flathub-export/export-flathub.py as a starting point.

**Comment by tsdgeos** [comment]
I just don't understand why you would want to do all that extra work just to end up in the same place we already are. Can you explain your rationale?

**Comment by Siosm** [comment]
With YAML we get comments and a simpler to write file format. The conversion will be a one time cost and in both cases, we need tools to automate updating to the latest release.

**Comment by Siosm** [comment]
This may not be the place to discuss this but my end goal is to have as many KDE apps as possible available and to push updates easily for each releases.

**Comment by tsdgeos** [comment]
You are correct, this is not the place to discuss this.

**Comment by Siosm** [comment]
bot, build org.kde.okteta

**Comment by Siosm** [comment]
bot, build org.kde.okteta

**Comment by nedrichards** [comment]
OK, so this is blocked by the upstream community discussion in https://github.com/flathub/flathub/pull/1579#issuecomment-645572347 the strength of KDE software on Flathub is its consistency and alignment with upstream.

**Comment by Siosm** [comment]
Yes, let's not merge more of those apps until we all agree on how to proceed. I'm working on a proposal.

**Comment by nedrichards** [comment]
Sorry, I didn't see this comment at the time and had literally just merged two of the other MRs. It's "ok" as they're JSON ones and can be supported by the same methods (like all KDE software I'll make sure all the KDE team has write access) but absolutely we should all move forward as one.

**Comment by Siosm** [comment]
Please do not merge, this is not yet ready.

**Comment by Siosm** [comment]
bot, build org.kde.okteta

**Comment by Siosm** [comment]
I have reverted this one to JSON and it should be OK to merge. @tsdgeos ?

**Comment by tsdgeos** [comment]
Let's go for it. @nedrichards @bilelmoussaoui can you merge it?

**Pull request closed without merging**